WordPress/gutenberg
# Show Template: Refactor rendering mode management in editor provider and store to persist it

Fixes: #68250 

## What?
Implements Data Persistence into Preferences Store for `Show Template`.

## Why?
Brings an enhancement to the previous implementation, where, the value set by the user for `Show Template` would get overwritten by subsequent refresh made to the page.

## How?
The store has been updated to consider using `preferences store` to persist this data.

## Testing Instructions
1. Create a new page from the `site editor`.
2. Change the value from `Show Template` toggle.
3. Refresh, and observe that the value now persists.

## Screencast
![Screen Recording Dec 24 2024](https://github.com/user-attachments/assets/41f25485-ee43-4425-9ca1-59037a457351)
